coordinator/src/node/rollover.rs

```diff
@@ -165,7 +163,7 @@ impl Rollover {
 
     /// Calculates the maturity time based on the current expiry timestamp.
     pub fn maturity_time(&self) -> OffsetDateTime {
-        coordinator_commons::calculate_next_expiry(self.expiry_timestamp, self.network)
+        coordinator_commons::calculate_next_expiry(OffsetDateTime::now_utc(), self.network)
```

Comment:
I'm not sure I follow this change. We should not always rollover, but only allow rollover from the old expiry to the next. Otherwise a user could extend an already expired contract. 

Also, looking at the whole logic: shouldn't be there a check somewhere in the `propose_rollover()` function if a rollover is still possible or if the position has already expired?

Reply:
> I'm not sure I follow this change. We should not always rollover, but only allow rollover from the old expiry to the next. Otherwise a user could extend an already expired contract. 

The commit message may be cryptic, but we are calculating the next expiry time here. Obviously we need to take the current date for calculating that, because if we use the current expiry we would always get the next expiry and not the closest to the current time.

> Also, looking at the whole logic: shouldn't be there a check somewhere in the `propose_rollover()` function if a rollover is still possible or if the position has already expired?


Yes, that check is basically comparing if the current expiry is the same on the position, If not propose rollover if it is then we have nothing to do and the position was just created in the rollover window or has already been rolled over.

Reply:
> The commit message may be cryptic, but we are calculating the next expiry time here. Obviously we need to take the current date for calculating that, because if we use the current expiry we would always get the next expiry and not the closest to the current time.

> Or in other words always get newer expiry and we only want the next from the current time.

Ah, that makes sense: we always want to get the _next_ expiry relative to _now_.

Reply:
> Yes, that check is basically comparing if the current expiry is the same on the position, If not propose rollover if it is then we have nothing to do and the position was just created in the rollover window or has already been rolled over.

But what if the position has already been expired? Am I missing this check?

Reply:
Look into the check if we need to rollover 😉 (on `rollover.rs`)

Reply:
Oh, I'm a blind 🦄. Thanks. 

```
 if expiry_timestamp < OffsetDateTime::now_utc() {
            bail!("Cannot rollover an expired position");
        }
```